media-cdn/snippets.py

```diff
@@ -0,0 +1,217 @@
+        base64_key: signing key as a base64 encoded string.
+        expiration_time: expiration time as a UTC datetime object.
+
+    Returns:
+        Returns the Signed URL appended with the query parameters based on the
+        specified URL prefix and configuration.
+    """
+    stripped_url = url.strip()
+    parsed_url = urllib.parse.urlsplit(stripped_url)
+    query_params = urllib.parse.parse_qs(
+        parsed_url.query, keep_blank_values=True)
+    encoded_url_prefix = base64.urlsafe_b64encode(
+            url_prefix.strip().encode('utf-8')).decode('utf-8')
+    epoch = datetime.datetime.utcfromtimestamp(0)
+    expiration_timestamp = int((expiration_time - epoch).total_seconds())
+    decoded_key = base64.urlsafe_b64decode(base64_key)
+
+    policy_pattern = u'URLPrefix={encoded_url_prefix}&Expires={expires}&KeyName={key_name}'
+    policy = policy_pattern.format(
+            encoded_url_prefix=encoded_url_prefix,
+            expires=expiration_timestamp,
+            key_name=key_name)
+
+    digest = ed25519.Ed25519PrivateKey.from_private_bytes(
+        decoded_key).sign(policy.encode('utf-8'))
+    signature = base64.urlsafe_b64encode(digest).decode('utf-8')
+    signature = signature[:-2]  # to remove trailing '=' symbols
+    signed_url = u'{url}{separator}{policy}&Signature={signature}'.format(
+            url=stripped_url,
+            separator='&' if query_params else '?',
+            policy=policy,
+            signature=signature)
+    return signed_url
+# [END mediacdn_sign_url]
+
+
+# [START mediacdn_sign_cookie]
+def sign_cookie(url_prefix, key_name, base64_key, expiration_time):
+    """Gets the Signed cookie value for the specified URL prefix and configuration.
+
+    Args:
+        url_prefix: URL prefix to sign as a string.
+        key_name: name of the signing key as a string.
+        base64_key: signing key as a base64 encoded string.
+        expiration_time: expiration time as a UTC datetime object.
+
+    Returns:
+        Returns the Cloud-CDN-Cookie value based on the specified configuration.
+    """
+    encoded_url_prefix = base64.urlsafe_b64encode(
+            url_prefix.strip().encode('utf-8')).decode('utf-8')
+    epoch = datetime.datetime.utcfromtimestamp(0)
+    expiration_timestamp = int((expiration_time - epoch).total_seconds())
+    decoded_key = base64.urlsafe_b64decode(base64_key)
+
+    policy_pattern = u'URLPrefix={encoded_url_prefix}:Expires={expires}:KeyName={key_name}'
+    policy = policy_pattern.format(
+            encoded_url_prefix=encoded_url_prefix,
+            expires=expiration_timestamp,
+            key_name=key_name)
+
+    digest = ed25519.Ed25519PrivateKey.from_private_bytes(
+        decoded_key).sign(policy.encode('utf-8'))
+    signature = base64.urlsafe_b64encode(digest).decode('utf-8')
+    signature = signature[:-2]  # remove trailing '='
+
+    signed_policy = u'Cloud-CDN-Cookie={policy}:Signature={signature}'.format(
```

Comment:
`Edge-Cache-Cookie` not `Cloud-CDN-Cookie`

Reply:
Updated in the latest code!